qiskit/assembler/assemble_schedules.py

```diff
@@ -82,6 +89,9 @@ def assemble_schedules(schedules, qobj_id, qobj_header, run_config):
             'instructions': qobj_instructions
         })
 
+    # set number of memoryslots
+    qobj_config['memory_slots'] = memory_slots or max_memory_slot
```

Comment:
Can we remove the `memory_slots` var? It's only used once on this line, but it's defined far from here. By the time the reader gets here, we have to scroll back up for the def

```suggestion
    qobj_config['memory_slots'] = qobj_config.pop('memory_slots', None) or max_memory_slot
```

In the above, I think it's more clear that we are comparing the config value to the value that we found in the instructions

Reply:
Yes, or remove `memory_slots` period as per my above comment.

Reply:
I removed the ability to set memory slots, it will now be inferred from the schedule in a similar way to quantum circuit inferring this from the classical registers.